toc/rfc/rfc-draft-pcap-bosh.md

```diff
@@ -2,60 +2,214 @@
+ 
+`pcap-release` is a bosh-release in development by the [App Runtime Platform WG](https://github.com/cloudfoundry/community/blob/main/toc/working-groups/app-runtime-platform.md). 
+
+The main features and noteworthy design choices are as follows:
+* Communication between the components is done via gRCP bidirectional streams. This allows sending capture requests and control messages (e.g. graceful stop), as well as sending status information about the capture in addition to the captured packet data.
+* Overloads of individual components can be mitigated or avoided automatically and via configuration:
+  * Components have **configurable limits for concurrent captures** (per source IP and overall)
+  * A **congestion control mechanism**, based on buffer fill states, notifies downstream components about the fact that not all the captured traffic can be forwarded. This is an indication for the user to e.g. adjust their capturing filter. At the same time this avoids flooding memory or network links accidentally.
+* Comprehensive logging of requests, connection issues, congestion and successful completion.
+* mTLS encrypted communication between all components, using gRPC over HTTP/2
+* Authentication and authorization are handled via the existing BOSH UAA and respective BOSH CLI tokens. Accordingly, the current solution is limited to using UAA and token based authentication with RSA tokens. Other authentication mechanisms can be implemented but require additional work.
+
+While the pcap-release is complex, it is also very flexible and versatile, with many advanced use cases that go beyond simply capturing network traffic to a client machine. While these benefits are not realized with the current state of the implementation, they are on the horizon. 
+
+The largest short-term benefit of integrating pcap-release with BOSH is the increased user base and feedback channel that will help improve pcap-release for the mid- and long-term use and future developments.
+
+### Add "pcap-lite", a thin front-end and merge functionality of tcpdump streams
 
-### Integrate PCAP CLI as a "pcap" keyword into BOSH CLI
+The alternative solution is based on the orchestration of existing tools with a small amount of new code.
 
-- This would allow PCAP BOSH CLI to share context with BOSH CLI and have access to the user's login session as well as the currently used BOSH environment.
+Akin to the BOSH SSH feature, a BOSH pcap feature is proposed with similar but slightly different workflow. The well-known `tcpdump` tool is launched on each VM via SSH and transmits its data via the SSH channel. In order to handle multiple VMs and merge their captures, multiple SSH sessions to the respective targets are opened in parallel, multiplexed in the bosh-cli and available for writing to disk or an output stream.
```

Comment:
```suggestion
Akin to the *BOSH SSH* feature, a *BOSH pcap* feature is proposed with a similar but slightly different workflow. The well-known `tcpdump` tool is launched on each VM via SSH and transmits its data via the SSH channel. In order to handle multiple VMs and merge their captures, multiple SSH sessions to the respective targets are opened in parallel, demultiplexed in the bosh-cli and available for writing to disk or an output stream.
```

Reply:
Agreed to use "merged", which will not cause any confusion.